src/layout/rect.rs

```diff
@@ -193,10 +193,44 @@ impl Rect {
             .try_into()
             .expect("invalid number of rects")
     }
+
+    /// Clamp the rect to fit inside the given rect.
+    ///
+    /// If the width or height of the rect is larger than the given rect, it will be clamped to
+    /// the given rect's width or height.
```

Comment:
Looking at this it is very unclear to me which rect you're talking about. Maybe reference to them as `self` and `parameter`.

Reply:
Makes sense - will change it to `this` and `other`

Reply:
fixed